src/helperFunctions.js

```diff
@@ -11,3 +11,45 @@ export const checkErrorMessage = (error, message) => {
     notify.show(message, 'error', 3000);
   }
 };
+export const mapOrder = (arrayToSort, arrayToOrderBy, keyToSortBy) => {
+  arrayToSort.sort((a, b) => {
```

Comment:
Before I run the function when the sources are returned, I add them to the end of the list  [here](https://github.com/openmrs/openmrs-ocl-client/pull/559/files#diff-e812e5fb3d2d789b04b88c4cfc2aefb5R84) so that the sources are sorted according to the list provided by Andrew [here](https://talk.openmrs.org/t/developing-the-ocl-for-openmrs-application/17771/1230?u=judeatu) then the rest that may not be in that list come last. And since this is a helper function that could be used again I didn't find it wise to add them directly here.

Reply:
I see. But if this is indeed a helper function, can we always assume that whoever uses it in the future will use a list which has all the expected items? I think it would be safer to account for the situation where both `indexOf`s return -1.